### PR TITLE
Issue 789

### PR DESCRIPTION
**Pull request by kameshsampath**
**MavenScmEnricher**
* to handle Maven SCM details ()
  * this will add the following annotations to the k8s resources
    * `fabric8.io/scm-con-url --> scm/connection`
    * ` fabric8.io/scm-devcon-url --> scm/developerConnection`
    *  `fabric8.io/scm-tag --> scm/tag`
    * ` fabric8.io/scm-url --> scm/url`

**MavenIssueTrackerEnricher**
* to handle Maven IssueManagement details (MavenIssueTrackerEnricher)
  * this will add the following annotations to the k8s resources
    * `fabric8.io/issue-system--> issueManagement/system`
    * ` fabric8.io/issue-tracker-url --> issueManagement/url `

above details will be extracted from pom.xml if found

### CODE REVIEW

**Comment by jstrachan** [comment]
[merge]

File: enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/MavenScmEnricher.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    static final String SCM_URL = "fabric8.io/scm-url";
+
+    public MavenScmEnricher(EnricherContext buildContext) {
+        super(buildContext, ENRICHER_NAME);
+    }
+
+    @Override
+    public Map<String, String> getAnnotations(Kind kind) {
+        Map<String, String> annotations = new HashMap<>();
+        if (kind.isDeployOrReplicaKind() || kind.isService()) {
```
Comment (rhuss) [inline enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/MavenScmEnricher.java]: Why this restriction to servicres and deployments (and not pods or daemonsets) ?

The choice looks a bit arbitrary.

Comment (rhuss) [inline enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/MavenScmEnricher.java]: Same btw for the GitEnricher from where this probably has been taken over.

File: enricher/standard/src/test/java/io/fabric8/maven/enricher/standard/MavenIssueManagementEnricherTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+/**
+ * @author kameshs
+ */
+@RunWith(JMockit.class)
+public class MavenIssueManagementEnricherTest {
+
+    @Mocked
+    private EnricherContext context;
+
```
Comment (rhuss) [inline enricher/standard/src/test/java/io/fabric8/maven/enricher/standard/MavenIssueManagementEnricherTest.java]: just a hint: to make you live easier you could even mock down until the issue management itself:

```java
@Mocked 
IssueManagement issueManagement
....
new Expectactions() {{
   context.getProject();
   project.getIssueManagement();
   issueManagement.getUrl(); result = "...";
   issueManagement.getSystem();  result = "..."
}}
```

File: enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/MavenScmEnricher.java (hunk 1 above)
Comment (kameshsampath) [inline enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/MavenScmEnricher.java]: yes ;)

File: enricher/standard/src/test/java/io/fabric8/maven/enricher/standard/MavenIssueManagementEnricherTest.java (hunk 2 above)
Comment (kameshsampath) [inline enricher/standard/src/test/java/io/fabric8/maven/enricher/standard/MavenIssueManagementEnricherTest.java]: yeah thats cool! did not think on it .. will do it

**Comment by kameshsampath** [comment]
@rhuss - https://github.com/fabric8io/fabric8-maven-plugin/pull/831 has the docs for the same, please review and merge

**Pull request merged**